From cb11f14f3c34d180cd173dd984b4961be617a8eb Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Fri, 7 Feb 2014 12:12:03 +0200
Subject: [PATCH 099/466] dpaa_eth: include mac.h where required

Include mac.h where required.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: Id2e7c768df4faf1230b7e52cc3de6bc601a9184c
Reviewed-on: http://git.am.freescale.net:8181/8648
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Cristian-Constantin Sovaiala <Cristian.Sovaiala@freescale.com>
Reviewed-by: Ruxandra Ioana Radulescu <ruxandra.radulescu@freescale.com>
[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_1588.c        |    1 +
 drivers/net/ethernet/freescale/dpa/dpaa_eth.h         |    2 --
 drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c  |    2 +-
 drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c |    1 +
 drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c   |    1 +
 drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c  |    1 +
 drivers/net/ethernet/freescale/dpa/dpaa_eth_sysfs.c   |    1 +
 drivers/net/ethernet/freescale/dpa/dpaa_ethtool.c     |    1 +
 drivers/staging/fsl_dpa_offload/dpa_stats.c           |    1 +
 9 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_1588.c b/drivers/net/ethernet/freescale/dpa/dpaa_1588.c
index 514c045..9f66640 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_1588.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_1588.c
@@ -30,6 +30,7 @@
 #include "dpaa_eth.h"
 #include "dpaa_eth_common.h"
 #include "dpaa_1588.h"
+#include "mac.h"
 
 static int dpa_ptp_init_circ(struct dpa_ptp_circ_buf *ptp_buf, u32 size)
 {
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.h b/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
index c7ab652..6d29e5a 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
@@ -47,8 +47,6 @@
 #include "lnxwrp_fsl_fman.h"
 #include "fm_ext.h"
 #include "fm_port_ext.h" /* FM_PORT_FRM_ERR_* */
-
-#include "mac.h"		/* struct mac_device */
 #ifdef CONFIG_FSL_DPAA_ETH_DEBUGFS
 #include "dpaa_debugfs.h"
 #endif /* CONFIG_FSL_DPAA_ETH_DEBUGFS */
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
index c20a96b..0d59dc4 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
@@ -42,7 +42,7 @@
 #include "dpaa_eth.h"
 #include "dpaa_eth_common.h"
 #include "dpaa_1588.h"
-
+#include "mac.h"
 
 /* DPAA platforms benefit from hardware-assisted queue management */
 #define DPA_NETIF_FEATURES	(0)
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c
index c0482d6..b7cfad3 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c
@@ -51,6 +51,7 @@
 #include "dpaa_eth_common.h"
 #include "dpaa_eth_base.h"
 #include "lnxwrp_fsl_fman.h" /* fm_get_rx_extra_headroom(), fm_get_max_frm() */
+#include "mac.h"
 
 /* For MAC-based interfaces, we compute the tx needed headroom from the
  * associated Tx port's buffer layout settings.
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
index 7229bb4..7ead9ff 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
@@ -45,6 +45,7 @@
 #include "dpaa_eth_common.h"
 #include "dpaa_eth_base.h"
 #include "lnxwrp_fsl_fman.h" /* fm_get_rx_extra_headroom(), fm_get_max_frm() */
+#include "mac.h"
 
 #define DPA_DESCRIPTION "FSL DPAA Proxy initialization driver"
 
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c
index a3290a7..6fe017c 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c
@@ -50,6 +50,7 @@
 #include "dpaa_eth_common.h"
 #include "dpaa_eth_base.h"
 #include "lnxwrp_fsl_fman.h" /* fm_get_rx_extra_headroom(), fm_get_max_frm() */
+#include "mac.h"
 
 /* forward declarations */
 static enum qman_cb_dqrr_result __hot
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sysfs.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sysfs.c
index 5eb5f62..bf01677 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sysfs.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sysfs.c
@@ -35,6 +35,7 @@
 #include <linux/io.h>
 #include <linux/of_net.h>
 #include "dpaa_eth.h"
+#include "mac.h"		/* struct mac_device */
 #ifdef CONFIG_FSL_DPAA_1588
 #include "dpaa_1588.h"
 #endif
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_ethtool.c b/drivers/net/ethernet/freescale/dpa/dpaa_ethtool.c
index 2f65ed8..c9145d4 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_ethtool.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_ethtool.c
@@ -41,6 +41,7 @@
 #include <linux/string.h>
 
 #include "dpaa_eth.h"
+#include "mac.h"                /* struct mac_device */
 
 static char stat_gstrings[][ETH_GSTRING_LEN] = {
 	"tx-rx-64-frames",
diff --git a/drivers/staging/fsl_dpa_offload/dpa_stats.c b/drivers/staging/fsl_dpa_offload/dpa_stats.c
index ab205e5..97564f5 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_stats.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_stats.c
@@ -37,6 +37,7 @@
 #include <linux/of_platform.h>
 #include "lnxwrp_fm.h"
 #include "dpaa_eth.h"
+#include "mac.h"
 #include <linux/fsl_qman.h>
 
 /* DPA offloading layer includes */
-- 
1.7.10.4

